src/index.d.ts

```diff
@@ -211,3 +225,10 @@ export class Immer {
      */
     setUseProxies(useProxies: boolean): void
 }
+
+export interface ImmerState<T = any> {
+    parent?: ImmerState
+    base: T
+    copy: T
+    assigned: {[prop: string]: boolean}
```

Comment:
Not 100% sure about the `assigned` field. On one hand it feels like leaking internal details, that can also be determined by the other hooks. On the other hand, it might be super handy, so probably we can leave it.

Reply:
@mweststrate Yeah, it's just nice to reuse the data instead of recomputing it manually. :)

Reply:
I can imagine, let's leave it